src/actions/actions.ts

```diff
@@ -1960,6 +1960,19 @@ class ActionDeleteChar extends BaseCommand {
 }
 
 @RegisterAction
+class ActionDeleteCharWithDeleteKey extends ActionDeleteChar {
+  keys = ["<delete>"];
+
+  public async execCount(position: Position, vimState: VimState): Promise<VimState> {
+    if (vimState.recordedState.count !== 0) {
```

Comment:
It seems to me when I press `<delete>`, it goes to `execCount` with a 0 count (default value for an int).
In your BaseCommand, you have this line to set it to 1 if it's 0:
`let timesToRepeat = this.canBePrefixedWithCount ? vimState.recordedState.count || 1 : 1;`